lib/spack/spack/environment/environment.py

```diff
@@ -897,20 +947,9 @@ def _construct_state_from_manifest(self):
         )
         self.spec_lists[user_speclist_name] = user_specs
 
-        enable_view = env_configuration.get("view")
-        # enable_view can be boolean, string, or None
-        if enable_view is True or enable_view is None:
-            self.views = {default_view_name: ViewDescriptor(self.path, self.view_path_default)}
-        elif isinstance(enable_view, str):
-            self.views = {default_view_name: ViewDescriptor(self.path, enable_view)}
-        elif enable_view:
-            path = self.path
-            self.views = dict(
-                (name, ViewDescriptor.from_dict(path, values))
-                for name, values in enable_view.items()
-            )
-        else:
-            self.views = {}
+        # Process views, starting with the one in the environment since it
+        # takes precedence.
+        self._process_view(spack.config.get("view"))
```

Comment:
```suggestion
        self._process_view(spack.config.get("view", True))
```
Just to have the default hardcoded like we try to do for all config sections. Then we can delete the `if self.views == dict()` code block at the end of the `_process_view` method.

Reply:
Adding this works but cannot get rid of the `dict()` check at the end of `_process_view` because that code can be reached if the `env_view` consists of an empty dictionary (e.g., an empty `spack.util.spack_yaml.syaml_dict`).

Reply:
I'm not sure how I talked myself out of thinking that. Oh well. Too much code review.